extensions/data-workspace/src/dialogs/openExistingDialog.ts

```diff
@@ -48,6 +48,19 @@ export class OpenExistingDialog extends DialogBase {
 					this.showErrorMessage(constants.ProjectFileNotExistError(this._projectFile));
 					return false;
 				}
+
+				// workspace file should end in .code-workspace
+				const workspaceValid = this.workspaceInputBox!.value!.endsWith(constants.WorkspaceFileExtension);
+				if (!workspaceValid) {
+					this.showErrorMessage(constants.WorkspaceFileInvalidError(this.workspaceInputBox!.value!));
+					return false;
+				}
+
+				const workspaceParentDirectoryExists = await directoryExist(path.dirname(this.workspaceInputBox!.value!));
+				if (!workspaceParentDirectoryExists) {
+					this.showErrorMessage(constants.WorkspaceParentDirectoryNotExistError(this.workspaceInputBox!.value!));
+					return false;
+				}
```

Comment:
This is really close to what's over in newProjectDialog.ts.  Maybe combine them in DialogBase?